### PR TITLE
Update for reference docs 02-sign-in.mdx

### PR DESCRIPTION
**Pull request by mdp18**
Updated the Nhost reference docs to align with the E.164 formatting standard that is required to send out an OTP.

main docs: https://docs.nhost.io/platform/authentication/sign-in-with-phone-number-sms
reference docs: https://docs.nhost.io/reference/javascript/auth/sign-in

### CODE REVIEW

File: docs/docs/reference/docgen/javascript/auth/content/hasura-auth-client/content/02-sign-in.mdx [hunk 1]
```diff
@@ -53,8 +53,8 @@ nhost.auth.signIn({ email: 'joe@example.com' })
 
 ```ts
 // [step 1/2] Passwordless sign in using SMS
-nhost.auth.signIn({ phoneNumber: '001122334455' })
+nhost.auth.signIn({ phoneNumber: '+11233213123' })
```
Comment (szilarddoro) [inline docs/docs/reference/docgen/javascript/auth/content/hasura-auth-client/content/02-sign-in.mdx:56]: Could you please add these changes to `packages/hasura-auth-js/src/hasura-auth-client.ts`? Our markdown files are automatically generated from JSDoc comments.

**Review by szilarddoro — CHANGES_REQUESTED** [review]

File: docs/docs/reference/docgen/javascript/auth/content/hasura-auth-client/content/02-sign-in.mdx (hunk 1 above)
Comment (mdp18) [reply docs/docs/reference/docgen/javascript/auth/content/hasura-auth-client/content/02-sign-in.mdx:56]: yes will do that now. Sorry about that.

Comment (mdp18) [reply docs/docs/reference/docgen/javascript/auth/content/hasura-auth-client/content/02-sign-in.mdx]: @szilarddoro -- done!

**Comment by szilarddoro** [comment]
We need two more things, then we are good to go:
1. A changeset describing this change: `pnpm changeset` (patch version)
2. Auto-generated docs (`lint-staged` should take care of this if you add the changeset)

**Review by szilarddoro — APPROVED** [review]
Thanks @mdp18! 🙌🏼

**Comment by mdp18** [comment]
@szilarddoro -- thanks for adding the changes and autodoc. I'm on a business trip and don't have access to my coding environment.

**Pull request merged**